File: ClearBlazorTest/ClearBlazor/Components/Common/ElementSizeInfo.cs
namespace ClearBlazor
{
    public class ElementSizeInfo : IEquatable<ElementSizeInfo>
    {
        public double ElementX { get; set; }
        public double ElementY { get; set; }
        public double ElementWidth { get; set; }
        public double ElementHeight { get; set; }

        public bool Equals(ElementSizeInfo? other)
        {
            if (other == null)
                return false;

            if (ElementX == other.ElementX &&
                ElementY == other.ElementY &&
                ElementWidth == other.ElementWidth &&
                ElementHeight == other.ElementHeight)
                return true;

            return false;
        }
    }
}

File: ClearBlazorTest/ClearBlazor/wwwroot/ElementSizeInfo.js
window.GetElementSizeInfo = (element) => {
    var elementRect = element.getBoundingClientRect();
    var parentRect = element.parentNode.getBoundingClientRect();
    return {
        ElementX: elementRect.x, ElementY: elementRect.y, ElementWidth: elementRect.width, ElementHeight: elementRect.height
    }
}

window.GetElementSizeInfoById = (id) => {
    var element = document.getElementById(id);
    if (!element)
        return;
    var elementRect = element.getBoundingClientRect();
    return {
        ElementX: elementRect.x, ElementY: elementRect.y, ElementWidth: elementRect.width, ElementHeight: elementRect.height
    }
}

window.GetParentElementSizeInfoById = (id) => {
    var element = document.getElementById(id);
    if (!element)
        return;
    var parentNode = element.parentNode;
    if (!parentNode)
        return;
    var parentRect = parentNode.getBoundingClientRect();
    return {
        ElementX: parentRect.x, ElementY: parentRect.y, ElementWidth: parentRect.width, ElementHeight: parentRect.height
    }
}
